### ISSUE TITLE
gen_snapshot does not include vmservice_io::main for Debug and Release builds

### ISSUE THREAD

**Issue by sjindel-google** [dart-lang/sdk#33866]
`dart_bootstrap` does not have this issue.
Precisely, the `vmservice_io` library is never seen in `Precompiler::AddAnnotatedRoots` when compiling with `gen_snapshot`. 
@mraleph @cbracken

**Comment by sjindel-google** [comment]
This issue only affects the Dart 1 pipeline.

`gen_snapshot` explicitly loads any libraries referenced by the entry-points file. If no functions in `vmservice_io` are mentioned, the entire library is not loaded, and `AddAnnotatedRoots` never sees the annotated function.

The solution I've implemented in https://dart-review.googlesource.com/c/sdk/+/65143 amends `gen_bootstrap` to explicitly load `vmservice_io` through the artificial `dart:vmservice_io` URI which was previously used in Flutter's entry-points file. This URI mapping is set up when compiling debug/profile Flutter apps, but is not always available. If the URI is not bound, `gen_bootstrap` silently moves on without loading `vmservice_io`.

I've tested that this fixes the issue with the Flutter microbenchmarks, but I feel that this isn't the ideal solution, and more complications may arise when migrating other entry-points to the @pragma definitions.
